build-support/scripts/protobuf.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
```

Comment:
I moved all of the miscellaneous things not counting invoking `protoc` into this script and out of the makefile so the bash-in-makefile problems aren't there anymore.

Also now this script is in charge of discovering `*.proto` files and always triggering `proto-gen.sh` on them, so the `make --always-make proto` problem is gone.